### PR TITLE
Remove all jquery from timecard.js

### PR DESCRIPTION
**Pull request by Sgtpluck**
## Description

This builds on the work of #1156, which was close but not quite there in terms of rooting out all the jquery on timecard.js. I looked briefly into just updating that PR, but with the accessibleAutocomplete change, it had drifted enough away from main that that would have been more difficult. Thanks for the work @hbillings and @jadudm 

This change:
1. Removes all code related to localStorage. Very happy to discuss this if anyone has feelings about it; Conversations with @Jkrzy have led me to believe that the cost in reasoning and brittleness is far greater than the benefits we're currently getting.
2. Removes all jquery in timecard.js
3. Reorganizes the code somewhat.
4. Adds JSDOC documentation to make the existing functions a bit more understandable.
5. Moves the test file into an integration directory, leaving open the option for future unit tests. (I did not export any tests, though, so I didn't make any other test changes.

Open Considerations:
1. I didn't add more tests. I could do that! It will take a bit longer, as puppeteer is very fragile.

### CODE REVIEW

File: tock/tock/static/js/components/timecard.js [hunk 1]
```diff
@@ -1,336 +1,391 @@
...
+  accessibleAutocomplete.enhanceSelectElement({
+    showAllValues: true,
+    defaultValue: '',
+    selectElement: newEntry.querySelector('select'),
+    onConfirm: handleConfirm,
+  });
+
+  // Increment the TOTAL_FORMS
+  // FIXME
+  // Why is this needed? the post breaks if it's removed.
```
Comment (Jkrzy) [inline tock/tock/static/js/components/timecard.js:283]: Good question! It's a component of Django's Formsets which we're using to process the inbound N "Django form for a timecard entry". Specifically the management form. https://docs.djangoproject.com/en/2.2/topics/forms/formsets/#understanding-the-managementform

If we _don't_ update this when adding adding project lines, the response behavior get's a bit wonky when there are validation failures. 

I.e. `id_timecardobjects-TOTAL_FORMS = 2`. We submit a timecard with 3 lines that fails validation.  Only the first two project lines will be rendered in the response, the third will be lost.

Comment (Sgtpluck) [reply tock/tock/static/js/components/timecard.js:283]: ah, thank you for the explanation! i will update this documenting that so no one else has to scratch their head confused.

File: tock/tock/static/js/components/timecard.js [hunk 2]
```diff
@@ -1,336 +1,391 @@
...
+/** @function
+ * Pulls data from the form
+ * @name getFormData
+ * @returns {FormData}
+ *
+ * */
 function getFormData() {
-  return $('.entry').map((i, entry) => {
+  return Array.from(document.querySelectorAll('.entry')).map((entry, i) => {
     const markedForDeletion = $('.entry-delete input', entry).prop('checked');
```
Comment (Jkrzy) [inline tock/tock/static/js/components/timecard.js:18]: Oooo, should this be a selector?

Comment (Sgtpluck) [reply tock/tock/static/js/components/timecard.js:18]: whoopsie, thank you for catching.

**Review by Jkrzy — APPROVED** [review]
:amaze: This is amazing, thanks @Sgtpluck!

**Pull request merged**